File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.kt
/**
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */

class Solution {
    fun removeNthFromEnd(head: ListNode?, n: Int): ListNode? {
        var current = head
        var size = 0
        while(current != null) {
            size++
            current = current.next
        }
        
        val index = size - n
        val dummy = ListNode(-1)
        dummy.next = head
        var prev: ListNode? = dummy
        current = dummy.next
        var c = -1
        while(c < index - 1) {
            prev = current
            current = current?.next
            c++
        }

        prev?.next = current?.next

        return dummy.next
    }
}

File: Data Structures & Algorithms/arranging-coins/submission-1.cpp
class Solution {
   public:
    int arrangeCoins(int n) {
        if (n == 0) {
            return 0;
        }
        if (n == 1) {
            return 1;
        }
        int count = 0;
        int coins = n;
        int index = 1;
        while (coins >= index) {
            coins -= index;
            index++;
            if (coins > 0) {
                count++;
            }
        }
        return count;
    }
};